### PR TITLE
chore(network): Replace use `DEFAULT_DISCOVERY_PORT` constant

### LINKED ISSUE #4274: Use the `DEFAULT_DISCOVERY_PORT` constant when possible
### Describe the feature

At the moment, in various places of the codebase, `port 30303` is hardcoded:

https://github.com/paradigmxyz/reth/blob/cbf3eb4333120cd48acf4be9533b1fb375daa842/crates/net/eth-wire/src/builder.rs#L103

There is a purpose built constant for that:

https://github.com/paradigmxyz/reth/blob/cbf3eb4333120cd48acf4be9533b1fb375daa842/crates/net/discv4/src/lib.rs#L97

Why not use it instead, except for unit tests where we want to be able to access some handmade harcoded values?

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by PatStiles**
Closes #4274

### CODE REVIEW

File: crates/net/discv4/src/lib.rs [hunk 1]
```diff
@@ -711,7 +711,7 @@ impl Discv4Service {
             // (e.g. connectivity problems over a long period of time, or issues during initial
             // bootstrapping) so we attempt to bootstrap again
             self.bootstrap();
-            return
+            return;
```
Comment (gakonst) [inline crates/net/discv4/src/lib.rs:714]: pls lint

Comment (PatStiles) [reply crates/net/discv4/src/lib.rs]: @gakonst resovled!

File: crates/net/eth-wire/src/builder.rs [hunk 2]
```diff
@@ -100,7 +101,7 @@ impl HelloBuilder {
                 client_version: "Ethereum/1.0.0".to_string(),
                 capabilities: vec![EthVersion::Eth68.into()],
                 // TODO: default port config
```
Comment (tcoratger) [inline crates/net/eth-wire/src/builder.rs:103]: This PR solves the TODO comment that can be removed in my opinion.

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request closed without merging**